### PR TITLE
oc_check_query for NAs in placename or latitude/longitude

### PR DESCRIPTION
**Pull request by dpprdan**
As of recently, OpenCage throws a HTTP 400 ‘bad query’ error when the query is empty, so we cannot be as sneaky as in \#89 anymore.
I.e. now we need to make sure that we do not send empty queries (empty placename
or latitude/longitude) to the API, but alert the user beforehand. This is what happens with this PR.

An empty request throws an HTTP 400 error (shown with `oc_process()` here, but the same happens with `oc_forward()` and `oc_reverse()` without this fix):

``` r
library(opencage)
opencage:::oc_process("", return = "df_list")
#> Error: HTTP failure: 401
#> missing API key
```

Instead, `oc_check_query()` now throws an error before moving to `oc_process()`/calling the API:

``` r
oc_forward("")
#> Error: `placename` must not be NA or an empty string.
oc_reverse(NA_real_, NA_real_)
#> Error: `latitude` must not be NA.
```

I was a bit reluctant to let the whole call error out ‘just’ because a user provides an empty/NA placename/coordinates, but this is a) what the API effectively demands now and b) essentially the same as providing incorrect coordinates, e.g. outside the (-)90/(-)180 bounds:

``` r
oc_reverse(100, 200)
#> Error: Every `latitude` must be between -90 and 90.
```

### CODE REVIEW

File: tests/testthat/test-oc_check_status.R [hunk 1]
```diff
@@ -4,53 +4,59 @@
...
 key_403 <- "2e10e5e828262eb243ec0b54681d699a" # always returns a 403 responce
+key_429 <- "d6d0f0065f4348a4bdfe4587ba02714b" # always returns a 429 responce
 
-test_that("oc_check_status returns 402 error if quota exceeded", {
+test_that("oc_check_status returns 400 error if request is invalid", {
   skip_on_cran()
   skip_if_offline()
 
+  # Both shouldn't happen since we oc_check_query
   expect_error(
```
Comment (maelle) [inline tests/testthat/test-oc_check_status.R:14]: so do we need this test? (sorry if my question is naive)

**Review by maelle — APPROVED** [review]
Thanks a lot! :rocket:

File: tests/testthat/test-oc_check_status.R (hunk 1 above)
Comment (dpprdan) [reply tests/testthat/test-oc_check_status.R:14]: That depends on how you define "need", I guess 😉 . Strictly speaking, I suppose one of the tests in test-oc_check_status would suffice, because it is either status 200 or not. However, I'd prefer to test it like this, so we can see whether we get a 40x response where we expect it. Only loosely related, but I wasn't expecting getting 400 errors now with empty queries which made the changes in this PR necessary. 
So long story short: If we want to keep things very concise, we do not need this test, but I don't think it hurts. (and no, not naïve at all). 😄

Comment (maelle) [reply tests/testthat/test-oc_check_status.R:14]: :+1: thanks for the explanation! And feel free to merge :-)
